File: 2.Algorithm/06.Dijkstra/배달/배달.java
import java.util.Arrays;

class Solution {
    public int solution(int N, int[][] road, int K) {
        int[] answer = new int[N];
        boolean[] visited = new boolean[N];
        Arrays.fill(answer,500001);
        Arrays.fill(visited,false);
        answer[0] = 0;
        int start = 0;
        while (true){
            visited[start] = true;
            for (int[] i: road){

                if (i[0]-1 == start && visited[i[1]-1] == false){
                    answer[i[1]-1] = Math.min(answer[i[1]-1], answer[start] + i[2]);
                }else if(i[1]-1 == start && visited[i[0]-1] == false){
                    answer[i[0]-1] = Math.min(answer[i[0]-1], answer[start] + i[2]);
                }
                
            }
            int min = 500001;
            for (int i = 1; i < N; i++){
                if (visited[i] == false && min >= answer[i]){
                    start = i;
                    min = answer[i];
                }
            }
            if(min == 500001){
                break;
            }
        }
        return (int)Arrays.stream(answer).filter(x -> x <= K).count();
    }
}

File: 2.Algorithm/06.Dijkstra/배달.py
def solution(N, road, K):
    answer = {x:500001 for x in range(1,N+1)}
    visited = [False for x in range(0,N+1)]
    answer[1] = 0
    visited[0] = True
    start = 1
    while False in visited:
        temp = {y:answer[y] for y in  list(filter(lambda x: visited[x]==False, answer.keys()))}
        start = min(temp,key=temp.get)
        visited[start] = True
        for i in road:
            if i[0] == start and visited[i[1]] == False:
                answer[i[1]] = min(answer[i[1]], answer[start] + i[2])
            if i[1] == start and visited[i[0]] == False:
                answer[i[0]] = min(answer[i[0]], answer[start] + i[2])
    return [1 if x <= K else 0 for x in answer.values()].count(1)